internal/handlers/v1alpha1/source.go

```diff
@@ -25,6 +27,75 @@ func NewServiceHandler(sourceService *service.SourceService) *ServiceHandler {
 	}
 }
 
+// validateSourceData validates the source data using the source validation rules
+func validateSourceData(data interface{}) error {
+	v := validator.NewValidator()
+	v.Register(validator.NewSourceValidationRules()...)
+	return v.Struct(data)
+}
+
+// authorizeSourceAccess checks if the user has access to the source
+// returns the source if authorized, nil and an error response otherwise
+// authorizeSourceAccess is a single function that handles authorization for all operations
+// It takes a response type parameter to return the appropriate error response type
+func (s *ServiceHandler) authorizeSourceAccess(
+	ctx context.Context,
+	sourceID uuid.UUID,
+	action string,
+	responseType string,
+) (*model.Source, interface{}, bool) {
+	source, err := s.sourceSrv.GetSource(ctx, sourceID)
+	if err != nil {
+		switch err.(type) {
+		case *service.ErrResourceNotFound:
+			switch responseType {
+			case "get":
+				return nil, server.GetSource404JSONResponse{Message: err.Error()}, false
+			case "delete":
+				return nil, server.DeleteSource404JSONResponse{Message: err.Error()}, false
+			case "update":
+				return nil, server.UpdateSource404JSONResponse{Message: err.Error()}, false
+			case "inventory":
+				return nil, server.UpdateInventory404JSONResponse{Message: err.Error()}, false
+			default:
+				return nil, server.GetSource404JSONResponse{Message: err.Error()}, false
+			}
+		default:
+			switch responseType {
+			case "get":
+				return nil, server.GetSource500JSONResponse{}, false
+			case "delete":
+				return nil, server.DeleteSource500JSONResponse{}, false
+			case "update":
+				return nil, server.UpdateSource500JSONResponse{}, false
+			case "inventory":
+				return nil, server.UpdateInventory500JSONResponse{}, false
+			default:
+				return nil, server.GetSource500JSONResponse{}, false
```

Comment:
I am unsure I really understand the advantage of this switch if you need to cast the type later anyway?